Success, Error, Loading Changes & more

bchehraz opened the pull request.

bchehraz committed:
Adds material-ui/lab dependency + fix query bug

bchehraz committed:
Adds useQuery logic for BookInfoPage + adds loading and error handling

bchehraz committed:
Adds useMutation to SignUpPage. Adds success and error logic, repositioned loading

bchehraz committed:
Deleted some unused request files

bchehraz committed:
Changed some HeaderSearchBox behavior

bchehraz committed:
Adds redirect after 1s after successfully uploading book. Repositions error/success position, adds Alert to snackbar

bchehraz committed:
Adjusts time to redirect after success to 1s from 1.5s in Login and Signup Pages

bchehraz committed:
Added onKeyDown for cases where useEffect isn't run perfectly and after you are done typing, a new search query may need to rerun

git-michelle reviewed (commented):
It looks like you found a way to reposition the snack bar which is great. 
I was trying to avoid using material-ui alert as it is from the lab and might have bugs or get removed https://material-ui.com/components/about-the-lab/

bchehraz committed:
Adds joinWaitlist check- Can't join waitlist if you are the one who has checked out the book

bchehraz committed:
Replaces Lab/Alert with reusable Alert component which takes severity and children as props

bchehraz committed:
Removes hardcoded signup input

bchehraz committed:
DashboardPage component stays in place if you refresh a subdirectory of dashboard

bchehraz committed:
Changes redirect from '/' to /dashboard

bchehraz committed:
Fixes success/error message severity

bchehraz committed:
Adds useQuery and getUser call in Dashboard with loading and error catching

bchehraz commented:
Client Updates:

- Added a custom "Alert" component which takes severity as prop. For some reason when passing this into a Snackbar, it must be wrapped with a div otherwise it breaks.
- Removed `@material-ui/lab` as a dependency
- Added `clsx` (small) dependency. You can see it in use in `screens/DashboardPage.js` and `components/Alert.js`
- Login and Signup will redirect to /dashboard now
- DashboardPage:
  - stays in place if page is refreshed. Will likely run into merge conflict once Michelle's branch is merged.
  - now has useQuery with getUser call.
  - has displayName being displayed from auth.user.displayName
  - has correct number of Inventory Items, Checked Out items, and Waitlisted items, instead of a "#". And displays a persistent spinner in place if there's an error, not just when it's loading.
  - has error snackbar in the top right

Server Updates:
- joinWaitlist now checks if the user is the one who has checked out the book and throws an error if so.

bchehraz committed:
Merge branch 'main' into bchehraz

bchehraz committed:
Routing Update: PrivateRoute moved to components. One route now handles all subdirectories in App.js. DashboardPage links updated

bchehraz committed:
Adds logic to handle tokenExpired error in context and useQuery+useMutation

bchehraz committed:
Refactored Login logic to use auth.login. LoginPage now shows expired token error if you've been redirected there for that reason

bchehraz committed:
Adds onTokenExpired to handle token expiration

bchehraz committed:
Fixes CheckedOut and Waitlist total values
